lib/main.js

```diff
@@ -205,4 +219,120 @@ module.exports = class Client {
             `Asset build server responded with unknown error. Http status ${statusCode}. Original message: "${message}".`
         );
     }
+
+    writer(type, files, options) {
+        if (type === 'css') {
+            return new CssWriter(files, options);
+        } else {
+            const writer = new JsWriter(files, options);
+
+            this.transforms.forEach(entry => {
+                writer.transform(entry.transform, entry.options);
+            });
+
+            this.plugins.forEach(entry => {
+                writer.plugin(entry.plugin, entry.options);
+            });
+
+            return writer;
+        }
+    }
+
+    determineType(values) {
+        return extname((values && values[0]) || '').replace('.', '');
+    }
+
+    async publishAssets(tag, entrypoints, options) {
+        Joi.assert(
```

Comment:
these are `Boom`ed into 400 errors?

Reply:
Annoyingly looks like no. 

A. Should we convert? 
These validations are intended to try to catch developer mistakes rather than runtime errors though I dare say they probably end up doing both...

B. Whats the best way? 
One option is to try/catch and Boom.boomify them. Not pretty...

Reply:
This is the client so I don't really see the benefit of boom errors here. Boom errors are errors to flag http error status responses.

Reply:
the client is used within a webapp in our case. But we wrap with boom there, so I agree that this client shouldn't bother.

Should it have boom at all?

Reply:
Will any of these errors ever surface up to a http endpoint?

Reply:
No, don't think so.